corehq/apps/locations/sql_templates/get_location_fixture_ids_2.sql

```diff
@@ -0,0 +1,333 @@
+DROP FUNCTION IF EXISTS get_location_fixture_ids_2(TEXT, INTEGER[], BOOLEAN);
+
+CREATE FUNCTION get_location_fixture_ids_2(
```

Comment:
Treating this SQL function as a black box, my understanding is that it is used in two ways:

1. To retrieve the set of locations to include in the locations fixture:
    ```
    for location in locations:
        yield all direct ancestor locations
        find the ancestor location whose type is location.location_type.expand_from (defaulting to location.location_type)
        yield all direct children of this ancestor location
        do likewise for those children's children recursively
        stop when you hit the location.location_type.expand_to level (defaulting to the bottom of the tree)
    return the union of these sets
    ```
1. To retrieve the set of locations to use as case owners:
    ```
    for location in locations:
        yield all direct ancestor locations
        yield all direct children of this location (ie, the fallback behavior from above)
        do likewise for those children's children recursively
        stop when you hit the location.location_type.restrict_cases_to level (defaulting to the bottom of the tree)
    return the union of these sets
    ```
That to me sounds like the SQL function could be tasked with returning the union of the trees for each passed-in location, expanded from and bounded according to two parameters which vary, as per the following call signatures:
use case 1:
```
get_location_fixture_ids_2(domain, location_pks, expand_from_param='expand_from', expand_to_param='expand_to')
```
use case 2 (use the fallback expand_from behavior)
```
get_location_fixture_ids_2(domain, location_pks, expand_from_param=None, expand_to_param='restrict_cases_to')
```
Like, it seems like the function shouldn't need to know anything about how it's being used beyond how to figure out where to expand from and to.  Could even rename it to something like `get_location_family` or some such.

Overall, I think the focus should be on simplicity, so if this wouldn't actually be simpler, then disregard my thoughts entirely.  Would love to hear if that idea resonates with either of you, @AddisonDunn and @millerdev

Reply:
I like the idea of simplifying the function, but in the example you gave for "use case 2" I didn't find the parameters
```
expand_from_param=None, expand_to_param='restrict_cases_to'
```
to be very intuitive. This is not meant as a criticism of your idea. It probably has more to do with me not having a good understanding of what it means to "restrict cases" in this context.

Reply:
> the other settings, `expand_from_root` and `include_only` (these should have no effect with use case 2)

Ah yeah, I forgot about those.  That does sound substantially different, then.  I see in a different thread that you're talking about extracting the shared functionality and making this two different functions.  That sounds like a promising path.